mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala

```diff
@@ -154,7 +154,11 @@ object TestingUtils {
      */
     def absTol(eps: Double): CompareVectorRightSide = CompareVectorRightSide(
       (x: Vector, y: Vector, eps: Double) => {
-        x.toArray.zip(y.toArray).forall(x => x._1 ~= x._2 absTol eps)
+        if (x.size != y.size) {
```

Comment:
How about just `x.size == y.size && ...`?
I think this also needs to be done for the matrix ops below.